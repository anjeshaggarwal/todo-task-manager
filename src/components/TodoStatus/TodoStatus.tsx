import React from 'react'
import './TodoStatus.css'

interface IProps {
    remaining: number,
    total: number
}

const TodoStatus = (props: IProps) => {
    return <b id='todo-status' data-testid='todo-status'>{`Total todos remaining: ${props.remaining} out of ${props.total}.`}</b>
}

export default TodoStatus