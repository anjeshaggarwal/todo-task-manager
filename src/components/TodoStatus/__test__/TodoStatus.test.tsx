import ReactDOM from 'react-dom'
import TodoStatus from './../TodoStatus'
import { render, cleanup } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TodoStatus remaining={0} total={0} />, div)
})

it('renders correctly', () => {
    const { getByTestId } = render(<TodoStatus remaining={0} total={0} />)
    expect(getByTestId('todo-status')).toBeTruthy()
})

it('match snapshot', () => {
    const element = renderer.create(<TodoStatus remaining={3} total={9} />).toJSON()
    expect(element).toMatchSnapshot()
})