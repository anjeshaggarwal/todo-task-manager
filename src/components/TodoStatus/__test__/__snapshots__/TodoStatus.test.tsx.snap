// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<b
  data-testid="todo-status"
  id="todo-status"
>
  Total todos remaining: 3 out of 9.
</b>
`;
